Apto/Catalog/Domain/Core/Factory/RuleFactory/Rule/Payload/RulePayloadFactory.php

```diff
@@ -23,16 +23,21 @@ public function __construct(ComputedProductValueCalculator $computedProductValue
 
     /**
      * @param ConfigurableProduct $product
-     * @param State $state
+     * @param State               $state
+     * @param bool                $indexedById
+     *
      * @return RulePayload
      * @throws \Apto\Base\Domain\Core\Model\InvalidUuidException
      * @throws \Apto\Catalog\Application\Core\Service\ComputedProductValue\CircularReferenceException
      */
-    public function getPayload(ConfigurableProduct $product, State $state)
+    public function getPayload(ConfigurableProduct $product, State $state, bool $indexedById = true)
     {
         return new RulePayload(
-            $this->computedProductValueCalculator->calculateComputedValuesByProduct($product->getProduct(), $state, true)
+            $this->computedProductValueCalculator->calculateComputedValuesByProduct($product->getProduct(), $state, $indexedById)
+
+            // todo maybe use this?
```

Comment:
Why should we use this? If we use this there will be one more call to fetch the product from repository. I think its not necessary if we already have the product aggregate